src/study.js

```diff
@@ -0,0 +1,66 @@
+async function init() {
+  const elements = [...document.querySelectorAll("[data-i18n-message]")];
+  for (const element of elements) {
+    element.innerHTML = DOMPurify.sanitize(browser.i18n.getMessage(element.dataset.i18nMessage), {
+      ALLOWED_TAGS: ["a"],
+      ALLOWED_ATTR: ["href"],
+    });
+  }
+
+  const brandStrings = await browser.experiments.settings.getStrings(["brandShortName", "brandFullName"]);
+
+  function createNodes(el, tagConfig) {
+    for (const childEl of [...tagConfig.childNodes]) {
+      if (childEl.nodeName === "#text") {
+        // Ignore whitespace nodes
+        if (!/[^\W]/.test(childEl.textContent)) {
+          continue;
+        }
+        const textOutput = childEl.textContent;
+        el.appendChild(document.createTextNode(textOutput));
+      } else {
+        el.appendChild(createElement(childEl));
+      }
+    }
+    return el;
+  }
+
+  function createElement(tagConfig) {
+    const el = document.createElement(tagConfig.tagName);
+    for (const attr of [...tagConfig.attributes]) {
+      el[attr.name] = attr.value;
+    }
```

Comment:
you could just call `.cloneNode(false)` instead, right?

Reply:
Yup :+1: